app/src/chainlit.py

```diff
@@ -1,32 +1,61 @@
 import logging
+from urllib.parse import parse_qs, urlparse
 
 import chainlit as cl
-import src.adapters.db as db
-from src.format import format_guru_cards
-from src.generate import generate
+from src import chat_engine
 from src.login import require_login
-from src.retrieve import retrieve
-from src.shared import get_embedding_model
 
 logger = logging.getLogger(__name__)
 
 require_login()
 
 
-@cl.on_message
-async def main(message: cl.Message) -> None:
-    logger.info(f"Received: {message.content!r}")
+@cl.on_chat_start
+async def start() -> None:
+    engine_id = engine_url_query_value()
+    logger.info("engine_id: %s", engine_id)
```

Comment:
```suggestion
    logger.info("Engine ID: %s", engine_id)
```
to make it human readable, or `logger.info(f"{engine_id=}")` for consistent formatting of variable name / value pairs